src/Components/Header/Header.js

```diff
@@ -24,19 +28,28 @@ export default function Header(){
         </div>
         <ul className="mr-5 gap-1">
             <NavLink to="/cart">
-                <li>
-                    <div className="badge-wrapper">
-                        <i className="bi bi-cart fa-lg"></i>
-                        <span className="icon-badge flex-align-center">2</span>
-                    </div>
-                </li>
-            </NavLink>
+            <li>
+                <div className="badge-wrapper">
+                    
+                    <i className="bi bi-cart fa-lg"></i>
+                    {isLogin ?
+                    <span className="icon-badge flex-align-center">
+                    {cart.item.length}</span>
+                    :
+                    <span></span> }
+                    
+                    
+                </div>
+            </li>
+        </NavLink>
+        
+            
             <NavLink to="/wishlist" >
                 <li>
                     <i className="bi bi-heart"></i>
                 </li>
             </NavLink>
-            {isLogin === true ? (
+            {isLogin === true && token ? (
```

Comment:
There is no need to explicitly check if "isLogin === true". "isLogin && token ?" works fine too.

Reply:
sure i will try it